src/sparseml/pytorch/sparsification/distillation/modifier_per_layer.py

```diff
@@ -208,28 +211,64 @@ def epsilon(self, value: float):
 
     @ModifierProp()
     def student_layer_names(self) -> List[str]:
+        """
+        :return: List of layer names to distill.
+        """
         return self._student_layer_names
 
     @student_layer_names.setter
     def student_layer_names(self, value: List[str]):
+        """
+        :param value: List of layer names to distill.
+            *Must be same length as teacher_layer_names*
+        """
         self._student_layer_names = value
 
     @ModifierProp()
     def teacher_layer_names(self) -> List[str]:
+        """
+        :return: List of layer names to distill
+        """
         return self._teacher_layer_names
 
     @teacher_layer_names.setter
     def teacher_layer_names(self, value: List[str]):
+        """
+        :param value: List of layer names to distill.
+            *Must be same length as student_layer_names*
+        """
         self._teacher_layer_names = value
 
     @ModifierProp()
     def project_features(self) -> bool:
+        """
+        :return: Whether to perform projection
+            between student and teacher layers
+        """
         return self._project_features
 
     @project_features.setter
     def project_features(self, value: bool):
+        """
+        :param value: Whether to perform projection
+            between student and teacher layers
+        """
         self._project_features = value
 
+    def state_dict(self) -> Dict[str, Dict]:
+        state = super().state_dict()
+        if self.project_features:
```

Comment:
we also need to save the optimizer group state dict here as well so they may be reloaded, right now, they're just left out from serialization

Reply:
that is what is saved in this block - the optimizer's extra group is made up of the self._projection state dict